lib/AST/FineGrainedDependencyFormat.cpp

```diff
@@ -382,11 +387,15 @@ void Serializer::writeMetadata() {
 }
 
 void
-Serializer::writeFineGrainedDependencyGraph(const SourceFileDepGraph &g) {
-  writeSignature();
-  writeBlockInfoBlock();
-
-  llvm::BCBlockRAII restoreBlock(Out, RECORD_BLOCK_ID, 8);
+Serializer::writeFineGrainedDependencyGraph(const SourceFileDepGraph &g,
+                                            bool forSwiftDeps) {
+  auto blockID = INCREMENTAL_INFORMATION_BLOCK_ID;
+  if (forSwiftDeps) {
+    writeSignature();
+    writeBlockInfoBlock();
+    blockID = RECORD_BLOCK_ID;
```

Comment:
Hmmm... seeing that the extra stuff is all at the start, maybe instead of passing in a bool, just have two separate functions??